textattack/shared/attack.py

```diff
@@ -55,7 +55,9 @@ def __init__(
         self.transformation = transformation
         if not self.transformation:
             raise NameError("Cannot instantiate attack without transformation")
-        self.is_black_box = getattr(transformation, "is_black_box", True)
+        self.is_black_box = (
+            getattr(transformation, "is_black_box", True) or search_method.is_black_box
```

Comment:
I think this should be an and, not an or. Because an attack with a white-box search method and black-box transformation is still a "white-box attack", right?

Reply:
Oh yeah you're right. Good catch.